### ISSUE TITLE
Codify fails when the input yaml object names have a dash in it.

### ISSUE THREAD

**Issue by fkautz** [kris-nova/naml#45]
Took the example stateless app from https://kubernetes.io/docs/tasks/run-application/run-stateless-application-deployment/ and ran it against naml.

```yaml
[...]
kind: Deployment
metadata:
  name: nginx-deployment
[...]
```
Ran this through naml:

```sh
kubectl get deployments -oyaml | naml c
Error during codify: unable to auto format code: 88:2: expected identifier on left side of :=2021-08-01T18:41:03-07:00 [Critical  ]  unable to auto format code: 88:2: expected identifier on left side of :=
```

I disabled gofmt to see the actual error and found this:

```go
nginx-deploymentDeployment := &appsv1.Deployment{
```

We should probably trim illegal characters, though this may in rare circumstances cause a duplicate name. e.g. two deployments on the same system with names `nginx-deployment` vs `nginxdeplolyment`

**Comment by fkautz** [comment]
Looks like we can safely rename all `-` characters in k8s object names to `_` in go. K8s does not allow `_` and go does not allow `-` in variable names.

**Comment by kris-nova** [comment]
pull requests accepted :)

https://github.com/kris-nova/naml/blob/main/codify/codify.go#L140

**Comment by fkautz** [comment]
working on it :)

**Comment by fkautz** [comment]
#46

**Comment by fkautz** [comment]
fixed in #46

**Comment by kris-nova** [comment]
closed with #46
released binaries with the patch https://github.com/kris-nova/naml/releases/tag/0.2.4
